platform/darwin/src/NSDictionary+MGLAdditions.mm

```diff
@@ -0,0 +1,51 @@
+#import "NSDictionary+MGLAdditions.h"
+
+#import "NSExpression+MGLAdditions.mm"
+
+@implementation NSDictionary (MGLAdditions)
+
+- (mbgl::PropertyMap)mbgl_propertyMap {
+    mbgl::PropertyMap propertyMap;
+    return [self propertyMapWithPropertyMap:propertyMap dictionary:self];
+}
+
+- (mbgl::PropertyMap)propertyMapWithPropertyMap:(mbgl::PropertyMap &)propertyMap dictionary:(NSDictionary *)dictionary {
```

Comment:
Passing in the property map by non-const reference is similar to an inout C parameter: you're actually mutating the map that's being passed in. So there's no need to return anything. In fact, this method could be renamed to "populatePropertyMap" or somesuch. But if an empty property map is always passed in, why not create it locally instead of accepting it as a parameter?